Backoffice/GeneratePerimeter/Strategy/NodeGeneratePerimeterStrategy.php

```diff
@@ -44,10 +44,9 @@ public function getType()
      */
     public function generatePerimeter()
     {
-        $treeNodes = $this->nodeRepository->findTreeNode($this->contextManager->getCurrentSiteId(), $this->contextManager->getUserCurrentSiteDefaultLanguage(), NodeInterface::ROOT_NODE_ID);
-        $treeNodes = $this->formatPerimeter($treeNodes[0]);
+        $treeNodes = $this->nodeRepository->findTreeNode($this->contextManager->getCurrentSiteId(), $this->contextManager->getUserCurrentSiteDefaultLanguage(), NodeInterface::ROOT_PARENT_ID);
 
-        return $treeNodes;
+        return $this->generateTreePerimeter(is_array($treeNodes) ? $treeNodes : array());
```

Comment:
nodeRepository->findTreeNode always return an array, event an empty one, so it seems this test is useless